airbyte-ci/connectors/pipelines/pipelines/publish.py

```diff
@@ -307,8 +307,10 @@ def reorder_contexts(contexts: List[PublishConnectorContext]) -> List[PublishCon
     Non strict-encrypt variant reference the strict-encrypt variant in their metadata file for cloud.
     So if we publish the non strict-encrypt variant first, the metadata upload will fail if the strict-encrypt variant is not published yet.
     As strict-encrypt variant are often modified in the same PR as the non strict-encrypt variant, we want to publish them first.
-    This is an hacky approach: as connector names with -strict-encrypt/secure prefix are longer,
-    they will be sorted first with our reverse sort below.
     """
 
-    return sorted(contexts, key=lambda context: context.connector.technical_name, reverse=True)
+    def has_secure_variant(context: PublishConnectorContext) -> bool:
+        SECURE_VARIANT_KEYS = ["secure", "strict-encrypt"]
+        return any(key in context.connector.technical_name for key in SECURE_VARIANT_KEYS)
```

Comment:
I think what was conusing for me is that this returns true for connectors that _are_ secure variants (source-file-secure) not a connector that have secure variants (source-file)

Outside of that naming I think this makes sense. If its a secure one, we reuturn true (1) which comes later than false (0) and then we reverse the list to get the secure variants first

Reply:
yep fair point on the naming - i'll rename in a follow up PR shortly